packages/smithy-aws-core/src/smithy_aws_core/credentials_resolvers/chain.py

```diff
@@ -0,0 +1,57 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  SPDX-License-Identifier: Apache-2.0
+from collections.abc import Sequence
+
+from smithy_core.aio.interfaces.identity import IdentityResolver
+from smithy_core.exceptions import SmithyIdentityException
+from smithy_core.interfaces.identity import IdentityProperties
+
+from smithy_aws_core.credentials_resolvers.environment import (
+    EnvironmentCredentialsSource,
+)
+from smithy_aws_core.credentials_resolvers.imds import IMDSCredentialsSource
+from smithy_aws_core.credentials_resolvers.interfaces import (
+    AwsCredentialsConfig,
+    CredentialsSource,
+)
+from smithy_aws_core.identity import AWSCredentialsIdentity, AWSCredentialsResolver
+
+_DEFAULT_SOURCES: Sequence[CredentialsSource] = (
+    EnvironmentCredentialsSource(),
+    IMDSCredentialsSource(),
+)
+
+
+class CredentialsResolverChain(
+    IdentityResolver[AWSCredentialsIdentity, IdentityProperties]
+):
+    """Resolves AWS Credentials from system environment variables."""
```

Comment:
This isn't just from Environment variables right? I'd expect this to describe something more about resolving an identity from a priority ordered list of credential sources.